src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -293,14 +330,14 @@ def get_dns_name_prefix(self, enable_validation: bool = False, **kwargs):
         read_from_mc = False
         if value_obtained_from_mc is not None:
             dns_name_prefix = value_obtained_from_mc
-            # clean up intermediate if `mc` has been decorated
-            self.remove_intermediate(parameter_name)
             read_from_mc = True
-        elif intermediate is not None:
-            dns_name_prefix = intermediate
         else:
             dns_name_prefix = raw_value
 
+        # skip dynamic completion & validation if option read_only is specified
+        if kwargs.get("read_only"):
```

Comment:
this is new thing?

Reply:
Yes, after removing the intermediates of all parameters, in order to reduce the introduction of additional content, we minimally add this option to a few getter functions to avoid state loss and loop calls.